consensus/istanbul/validator/weighted.go

```diff
@@ -752,43 +752,28 @@ func filterValidators(isSingleMode bool, govNodeAddr common.Address, weightedVal
 //  - []*weightedValidator : a list of validators which type is converted to weightedValidator
 //  - []float64 : a list of stakingAmounts.
 func getStakingAmountsOfValidators(validators istanbul.Validators, stakingInfo *reward.StakingInfo) ([]*weightedValidator, []float64, error) {
-	numValidators := len(validators)
-	weightedValidators := make([]*weightedValidator, numValidators)
-	stakingAmounts := make([]float64, numValidators)
-	addedStaking := make([]bool, len(stakingInfo.CouncilNodeAddrs))
+	nVals := len(validators)
+	weightedVals := make([]*weightedValidator, nVals)
+	stakingAmounts := make([]float64, nVals)
 
 	for vIdx, val := range validators {
 		weightedVal, ok := val.(*weightedValidator)
 		if !ok {
 			return nil, nil, errors.New(fmt.Sprintf("not weightedValidator. val=%s", val.Address().String()))
 		}
-		weightedValidators[vIdx] = weightedVal
-
-		sIdx, err := stakingInfo.GetIndexByNodeAddress(weightedVal.address)
-		if err == nil {
-			rewardAddr := stakingInfo.CouncilRewardAddrs[sIdx]
-			weightedVal.SetRewardAddress(rewardAddr)
-			stakingAmounts[vIdx] = float64(stakingInfo.CouncilStakingAmounts[sIdx])
-			addedStaking[sIdx] = true
-		} else {
-			weightedVal.SetRewardAddress(common.Address{})
-		}
-	}
+		weightedVals[vIdx] = weightedVal
```

Comment:
@hyunsooda It seems that `SetRewardAddress` calls disappear.  Don't we need to call them?

Reply:
Great finding! This lacks can forward block rewards to `REWARDBASE` account, not the registered reward address to the CnStaking contract. Updated